test/unit/test_c_api.cpp

```diff
@@ -6104,9 +6108,13 @@ TEST_F(CApi, open_plenty_of_contexts) {
     // database
     std::vector<FILE *> dummyFilePointers;
     std::vector<PJ_CONTEXT *> ctxts;
-    // 1024 is the number of file descriptors that can be opened simultaneously
-    // by a Linux process (by default)
-    for (int i = 0; i < 1024 - 50; i++) {
+    // The number of file descriptors that can be opened simultaneously by a
+    // process varies across platforms so we make use of getrlimit(2) to
+    // retrieve it.
+    struct rlimit open_max_limits;
+    getrlimit(RLIMIT_NOFILE, &open_max_limits);
+    const int open_file_max = static_cast<int> open_max_limits.rlim_cur;
```

Comment:
you need the parenthesis as in my initial suggestion:
```suggestion
    const int open_file_max = static_cast<int>(open_max_limits.rlim_cur);
```

Reply:
I've committed my suggestion

Reply:
Apologies.

Your suggestion on this made me realize I'd left the int declaration from the earlier sysconf call. We can just use rlim_t here.